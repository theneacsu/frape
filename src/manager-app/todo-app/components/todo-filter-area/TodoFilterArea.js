import React, { Component } from 'react'
import { connect } from 'react-redux'
import {
  setHideCompleted,
  setShowAll,
  setSearchTerm
} from '../../../../actions/todo-app/filters'
import styles from './TodoFilterArea.module.css'

export class TodoFilterArea extends Component {
  state = {
    showAll: true
  }

  onChangeCheckbox = e => {
    const checked = e.target.checked
    checked ? this.props.setHideCompleted() : this.props.setShowAll()
  }

  onChangeSearch = e => {
    const searchTerm = e.target.value
    this.props.setSearchTerm(searchTerm)
  }

  render() {
    return (
      <div className={styles.div}>
        <input
          className="input"
          type="text"
          placeholder="Search todos"
          onChange={this.onChangeSearch}
        />

        <input
          type="checkbox"
          className={styles.checkbox}
          id="completed-todos"
          onChange={this.onChangeCheckbox}
        />
        <label htmlFor="completed-todos">
          <span className={styles.span}>Hide completed</span>
        </label>
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  setSearchTerm: searchTerm => dispatch(setSearchTerm(searchTerm)),
  setHideCompleted: () => dispatch(setHideCompleted()),
  setShowAll: () => dispatch(setShowAll())
})

export default connect(
  undefined,
  mapDispatchToProps
)(TodoFilterArea)
