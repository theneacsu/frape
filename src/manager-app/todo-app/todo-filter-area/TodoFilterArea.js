import React from 'react'

const TodoFilterArea = () => (
  <div>
    <label for="completed-todos">Hide completed</label>
    <input type="checkbox" id="completed-todos" />
  </div>
)

export default TodoFilterArea
