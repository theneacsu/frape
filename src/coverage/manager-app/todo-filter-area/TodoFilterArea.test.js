import React from 'react'
import { shallow } from 'enzyme'
import TodoFilterArea from '../../../manager-app/todo-app/todo-filter-area/TodoFilterArea'

test('Should render correctly the TodoFilterArea component', () => {
  const wrapper = shallow(<TodoFilterArea />)
  expect(wrapper).toMatchSnapshot()
})
