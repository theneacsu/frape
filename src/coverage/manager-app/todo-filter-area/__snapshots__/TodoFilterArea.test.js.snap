// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly the TodoFilterArea component 1`] = `
<div>
  <label
    for="completed-todos"
  >
    Hide completed
  </label>
  <input
    id="completed-todos"
    type="checkbox"
  />
</div>
`;
